baaas-decision-fleet-manager/src/test/java/org/kie/baaas/dfm/app/controller/DecisionResourceValidationRestTest.java

```diff
@@ -0,0 +1,94 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.kie.baaas.dfm.app.controller;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.MediaType;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.kie.baaas.dfm.api.decisions.DecisionRequest;
+import org.kie.baaas.dfm.api.decisions.Model;
+import org.kie.baaas.dfm.app.manager.DecisionManager;
+import org.kie.baaas.dfm.app.manager.validators.ValidationTestProfile;
+import org.kie.baaas.dfm.app.model.DecisionVersion;
+import org.kie.baaas.dfm.app.storage.DMNStorageRequest;
+import org.kie.baaas.dfm.app.storage.DecisionDMNStorage;
+import org.kie.baaas.dfm.app.storage.s3.S3DMNStorage;
+import org.mockito.Mockito;
+
+import io.quarkus.test.junit.QuarkusMock;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+import io.quarkus.test.security.TestSecurity;
+
+import static io.restassured.RestAssured.given;
+import static org.kie.baaas.dfm.app.TestConstants.DEFAULT_CUSTOMER_ID;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.when;
+
+@QuarkusTest
+@TestProfile(ValidationTestProfile.class)
+public class DecisionResourceValidationRestTest {
+    @Inject
+    DecisionDMNStorage dmnStorage;
+
+    @BeforeAll
+    public static void beforeAll() {
+        S3DMNStorage storage = Mockito.mock(S3DMNStorage.class);
+        QuarkusMock.installMockForType(storage, S3DMNStorage.class);
+    }
+
+    private DMNStorageRequest createStorageRequest() {
+        DMNStorageRequest request = new DMNStorageRequest("provider-url", "hash");
+        when(dmnStorage.writeDMN(anyString(), Mockito.any(DecisionRequest.class), Mockito.any(DecisionVersion.class))).thenReturn(request);
+        return request;
+    }
+
+    private DecisionRequest createApiRequest() {
+        Model model = new Model();
+        model.setDmn("<xml><test>\"hello\"</test></xml>");
+
+        DecisionRequest decisions = new DecisionRequest();
+        decisions.setDescription("The Best Decision Ever");
+        decisions.setName("first-decision");
+        decisions.setModel(model);
+        decisions.setKind("Decision");
+        return decisions;
+    }
+
+    @Inject
+    DecisionManager decisionManager;
+
+    @TestSecurity(user = DEFAULT_CUSTOMER_ID)
+    @Test
+    void testDecisionRequestValidator_exceedAllowedLimit() {
+        //Max allowed decision is 1 for this test
+
+        //Mock a storage for DMN
+        createStorageRequest();
+
+        //creates the first decision
+        DecisionRequest apiRequest = createApiRequest();
+        decisionManager.createOrUpdateVersion(DEFAULT_CUSTOMER_ID, apiRequest);
+
+        //creates 2nd one through HTTP and fails at max allowed limit validator
+        given().when()
```

Comment:
@robpblake  
If I make two REST assured calls consecutively to create two Decisions in this same test method, the 2nd call can't pick up the decision created by the 1st call. Therefore, I directly create one by calling c.or.u.version method @L86.

Reply:
@myeung18 Can you explain what you mean when you say "2nd call can't view the decision created by the first call"? The API is returning no data?

In any case, pre-populating the database as part of the test setup is fine in my opinion using the `DecisionManager`.